pkgs/by-name/pi/pihole/package.nix

```diff
@@ -0,0 +1,259 @@
+{
+  lib,
+  fetchFromGitHub,
+  makeWrapper,
+  installShellFiles,
+  bash,
+  coreutils,
+  curl,
+  dig,
+  gawk,
+  getent,
+  glibc,
+  gnugrep,
+  gnused,
+  iproute2,
+  jq,
+  killall,
+  libidn2,
+  locale,
+  ncurses,
+  netcat,
+  nettools,
+  pihole-ftl,
+  procps,
+  resholve,
+  sqlite,
+  systemd,
+  util-linux,
+  stateDir ? "/etc/pihole",
+  ...
+}:
+
+(resholve.mkDerivation rec {
+  pname = "pihole";
+  version = "6.1";
+
+  src = fetchFromGitHub {
+    owner = "pi-hole";
+    repo = "pi-hole";
+    rev = "v${version}";
+    hash = "sha256-aEnv8Lhb5vf0yDyuriVTaUY1wcdVmTdqoK+KDHvT/Lw=";
+  };
+
+  patches = [
+    ./0001-Remove-sudo.patch
+    ./0002-Remove-unsupported-commands.patch
+    ./0003-Fix-redefinition-of-readonly-variable-utilsfile.patch
+  ];
```

Comment:
I see that the filenames are pretty self-explanatory already, but it would be a good idea to add a comment explaining why those patches are needed and if they need to be removed at the next bump.